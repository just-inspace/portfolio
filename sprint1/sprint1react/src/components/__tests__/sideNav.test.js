import { shallow, mount, render } from "enzyme";
import SideNav from "../sideNav.js";
import React from "react";

describe("SideNav", () => {
	it("renders correctly", () => {
		const wrapper = shallow(<SideNav />);
		expect(wrapper).toMatchSnapshot();
	});
});
