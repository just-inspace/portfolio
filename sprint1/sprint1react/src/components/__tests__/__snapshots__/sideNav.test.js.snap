// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          Home
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/projects"
        >
          Projects
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/resume"
        >
          Resume
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/contact"
        >
          Contact
        </NavLink>,
        <div
          id="sideNavControl"
        />,
      ],
      "id": "sideNav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Home",
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Projects",
          "to": "/projects",
        },
        "ref": null,
        "rendered": "Projects",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Resume",
          "to": "/resume",
        },
        "ref": null,
        "rendered": "Resume",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Contact",
          "to": "/contact",
        },
        "ref": null,
        "rendered": "Contact",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "sideNavControl",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            Home
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/projects"
          >
            Projects
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/resume"
          >
            Resume
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/contact"
          >
            Contact
          </NavLink>,
          <div
            id="sideNavControl"
          />,
        ],
        "id": "sideNav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Projects",
            "to": "/projects",
          },
          "ref": null,
          "rendered": "Projects",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Resume",
            "to": "/resume",
          },
          "ref": null,
          "rendered": "Resume",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Contact",
            "to": "/contact",
          },
          "ref": null,
          "rendered": "Contact",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "sideNavControl",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
